src/flag_gems/runtime/device/device.py

```diff
@@ -0,0 +1,51 @@
+import os
+import subprocess
+
+import torch
+import triton
+
+from .. import backend, error
+
+
+class device_ctx:
+    def __init__(self, vendor_name=None):
+        self.vendor_list = backend.vendors_map.keys()
+        self.device_info = self.get_vendor(vendor_name)
+        self.vendor_name, self.device_name, self.cmd, self.vendor = self.device_info
+
+    def get_vendor(self, vendor_name=None) -> tuple:
+        if vendor_name is not None:
+            return backend.get_vendor_info(vendor_name)
+        vendor_from_evn = self._get_vendor_from_evn()
```

Comment:
If you mean environment, please use env, which is a more accepted abbreviation for it.

Reply:
Get it.